### PR TITLE
feat(dedicated): filter organizations when order ipv4 block

### PR DESCRIPTION
**Pull request by mhelhali-soufien**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master` 
| Bug fix?         | yes/no
| New feature?     | yes/no
| Breaking change? | yes/no
| Tickets          | Fix #MANAGER-15314
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

### CODE REVIEW

File: packages/manager/apps/dedicated/client/app/components/ip/agoraOrder/ipv4/ipv4.constant.js [hunk 1]
```diff
@@ -67,6 +67,17 @@ export const BLOCK_ADDITIONAL_IP = 'Block Additional IP';
 
 export { TRACKING_PREFIX, ADDITIONAL_IP, ALERT_ID, IP_FAILOVER_PLANCODE };
 
+export const SERVER_REGION = {
+  USA: 'USA',
+  CANADA: 'CANADA - ASIA',
+  EUROP: 'EUROPE',
```
Comment (sachinrameshn) [inline packages/manager/apps/dedicated/client/app/components/ip/agoraOrder/ipv4/ipv4.constant.js:73]:
```suggestion
  EUROPE: 'EUROPE',
```

File: packages/manager/apps/dedicated/client/app/components/ip/agoraOrder/ipv4/ipv4.controller.js [hunk 2]
```diff
@@ -228,13 +230,13 @@ export default class AgoraIpV4OrderController {
   static getRegionFromServiceName(serviceName) {
     const serviceExt = last(serviceName.split('.'));
     if (['eu', 'net'].includes(serviceExt)) {
-      return 'EUROPE';
+      return SERVER_REGION.EUROP;
```
Comment (sachinrameshn) [inline packages/manager/apps/dedicated/client/app/components/ip/agoraOrder/ipv4/ipv4.controller.js:233]:
```suggestion
      return SERVER_REGION.EUROPE;
```

File: packages/manager/apps/dedicated/client/app/components/ip/agoraOrder/ipv4/ipv4.controller.js [hunk 3]
```diff
@@ -404,7 +407,29 @@ export default class AgoraIpV4OrderController {
...
+            case SERVER_REGION.CANADA:
+              registry = ORGANISATION_GROUP.ARIN;
+              break;
+            default:
+              registry = null;
+          }
+          this.organisations = registry
+            ? organisations.filter((org) => org.registry === registry)
+            : organisations;
+        } else {
```
Comment (sachinrameshn) [inline packages/manager/apps/dedicated/client/app/components/ip/agoraOrder/ipv4/ipv4.controller.js:430]: I guess we should't have a else case. This change is for all the service not only for dedicated server. We need to retain this for all service type.

**Review by sachinrameshn — CHANGES_REQUESTED** [review]

**Review by qpavy — DISMISSED** [review]

File: packages/manager/apps/dedicated/client/app/components/ip/agoraOrder/ipv4/ipv4.controller.js [hunk 4]
```diff
@@ -404,7 +407,22 @@ export default class AgoraIpV4OrderController {
...
+        switch (REGION) {
+          case SERVER_REGION.EUROPE:
+            registry = ORGANISATION_GROUP.RIPE;
+            break;
+          case SERVER_REGION.USA:
+          case SERVER_REGION.CANADA:
+            registry = ORGANISATION_GROUP.ARIN;
+            break;
+          default:
+            registry = null;
```
Comment (sachinrameshn) [inline packages/manager/apps/dedicated/client/app/components/ip/agoraOrder/ipv4/ipv4.controller.js:421]:
```suggestion
          case SERVER_REGION.EUROPE:
            registry = ORGANISATION_GROUP.RIPE;
            break;
          case SERVER_REGION.CANADA:
            registry = ORGANISATION_GROUP.ARIN;
            break;
          default:
            registry = null;
```

**Review by sachinrameshn — CHANGES_REQUESTED** [review]
I believe we need to show both the organisation for US region.

**Review by qpavy — APPROVED** [review]

File: packages/manager/apps/dedicated/client/app/components/ip/agoraOrder/ipv4/ipv4.controller.js [hunk 5]
```diff
@@ -372,9 +378,6 @@ export default class AgoraIpV4OrderController {
           .then((data) => {
             let countries = data;
```
Comment (sachinrameshn) [inline packages/manager/apps/dedicated/client/app/components/ip/agoraOrder/ipv4/ipv4.controller.js:379]: If at all we have countries coming from API how will we handling the organization filter ? 
We are dependent on region from service name if the country is coming as empty. And I see you have removed this logic. Shouldn't we need to keep as is and get the region based on countries ?

**Review by sachinrameshn — DISMISSED** [review]

**Review by qpavy — DISMISSED** [review]

**Pull request merged**